### PR TITLE
Fix shannon entropy

### PR DESCRIPTION
**Pull request by FanwangM**
The original implementation was not correct and this PR fixes the problem, #67.

### CODE REVIEW

**Comment by FanwangM** [comment]
The original usage of using Shannon entropy for molecule selection can be found in section 2.1 of https://pubs.acs.org/doi/abs/10.1021/ci900159f where we have a complete form of Shannon entropy,

```math
SE = \sum_i^m SE_i
```
where
```math
SE_i  =  - p_i \log_2{p_i }  - (1 - p_i)\log_2(1 - p_i)
```
and 
```math
p_i = \sum_j^m  x_{ij}
```

In Equation 2 of https://onlinelibrary.wiley.com/doi/10.1002/jcc.24423 in 2016, there is a scaling factor difference which ensures the value in the range of 0 and 1. This paper also introduced the Gini coefficient.

```math
SE = - \frac{ \sum \frac{y_i}{N} \ln {\frac{y_i}{N} } } {L \frac {\ln 2} {2}}
```

In a J Cheminform paper in 2021, https://doi.org/10.1186/s13321-021-00554-8. The definition was further simplified into 

```math
SE = \sum_i^n -p_i \log p_i
```
which only counts the on-bits (only 1 in the bit array).

File: DiverseSelector/diversity.py [hunk 1]
```diff
@@ -188,17 +197,9 @@ def logdet(x: np.ndarray) -> float:
     return f_logdet
 
 
-def shannon_entropy(x: np.ndarray) -> float:
+def shannon_entropy(x: np.ndarray, normalize=True, truncation=False) -> float:
```
Comment (FarnazH) [inline DiverseSelector/diversity.py:200]: @FanwangM, you forgot to add these two new arguments to the `Parameters` section.

**Review by FarnazH — COMMENTED** [review]
@FanwangM, extending the scope of the `shannon_entropy` function, shouldn't we remove the `entropy` function then?

**Comment by FanwangM** [comment]
> @FanwangM, extending the scope of the `shannon_entropy` function, shouldn't we remove the `entropy` function then?

Nice catch. I will do it soon.

File: DiverseSelector/diversity.py (hunk 1 above)
Comment (FanwangM) [reply DiverseSelector/diversity.py:148]: Fixed in [6e736ec](https://github.com/theochem/DiverseSelector/pull/177/commits/6e736ecd66b848a5e94c98b3d47be52870e11c30).

**Review by FarnazH — APPROVED** [review]

**Pull request merged**